src/DynamoCoreWpf/Views/Core/DynamoView.xaml.cs

```diff
@@ -222,7 +228,7 @@ internal TabItem AddExtensionTabItem(IViewExtension viewExtension, ContentContro
                 // creates a new tab item
                 tab = new TabItem();
                 tab.Header = viewExtension.Name;
-                tab.Tag = viewExtension.GetType();
+                tab.Tag = viewExtension;
```

Comment:
Can we consider if this causes any memory leak issues?

Reply:
I don't see how to be honest. Did you see a problem?

Reply:
Not really - tab's lifetime should be the same as the extension in most cases, maybe shorter.
In a testing scenario I could see it being an issue if the UI is already leaking, and as a consequence tab is never cleaned up, now viewExtension will not be cleaned up either.

Reply:
If we are leaking tabs we are probably also leaking the controls within them. The `viewExtension` object might add a bit more, but I'd say most of the harm would already be done.